zookeeper-api/src/test/java/org/apache/helix/zookeeper/impl/client/TestSharedZkClient.java

```diff
@@ -66,4 +65,16 @@ public void testCreateEphemeralFailure() {
       // this is expected.
     }
   }
+
+   /*
+   Tests clean up of closed connection from pool
+   Use cases tested
+     -Not removal of connection when there is a active watchers
+     -Removal of connection when there are no active watchers
+   */
+   @Test
+    public void testCleanupOfClosedConnectionFromConnectionManager(){
```

Comment:
Please finish the test : )

Reply:
Actually, am bit stuck with how to test this and i was wondering if someone can guide me here.  I posted it here - https://github.com/apache/helix/pull/1292#issuecomment-680828563

Reply:
One way that I can think of is modifying the test only method getActiveConnectionCount() so it can return either the total connection manager count or active manager count according to an input parameter.
My only concern is that I don't like the idea of making these test methods more complicated : ( However, it might be the easiest way.

Please try to think of any better idea. If we really don't have anything better, upgrading getActiveConnectionCount is still an OK option.